File: 11/CG_IntersectHalfplanesDll/Primitives/Segment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CG_IntersectHalfplanesDll.Primitives {
    public class Segment {
         private readonly Point left;
     private readonly Point right;
 
        public Segment(Point left, Point right) {

        if(left.getX() > right.getX()) {
            Point temp = right;
            right = left;
            left = temp;
        }

        left.setSegment(this);
        right.setSegment(this);

        this.left = left;
        this.right = right;
    }

    public Point getLeft() {
        return left;
    }

    public Point getRight() {
        return right;
    }

    public Point getBisectionalPoint() {
        return new Point((left.getX() + right.getX()) / 2,
                (left.getY() + right.getY()) / 2);
    }

    public Line getLine() {
        return new Line(left, right);
    }


        public override bool Equals(object o2) {
              if (! (o2 is Segment)) {
            return false;
        }
             Segment s2 = (Segment) o2;
             return ((this.getLeft().Equals(s2.getLeft()) &&
                this.getRight().Equals(s2.getRight())) ||
                (this.getLeft().Equals(s2.getRight()) &&
                        this.getRight().Equals(s2.getLeft())));
        }

      

       

       
    }

    
}

File: pr5/src/Segment.java
import java.awt.*;

/**
 * Created by Lev on 22.02.14.
 */
public class Segment {
    private Point left;
    private Point right;
    private Color color;
    public Segment(Point left, Point right) {
        this.left = left.getX() > right.getX()? right : left;
        this.right = left.getX() > right.getX()? left : right;
    }
    public Point getLeft() {
        return left;
    }
    public Point getRight() {
        return right;
    }
    public void setColor(Color color) {
        this.color = color;
    }
    public Color getColor() {
        return color;
    }
    public boolean equal(Segment op) {
        return left.equal(op.left) && right.equal(op.right);
    }
}
